core/src/stage.cpp

```diff
@@ -103,6 +103,7 @@ bool StagePrivate::storeSolution(const SolutionBasePtr& solution) {
 			return false;  // drop solution
 		failures_.push_back(solution);
 	} else {
+		solution->setComputeTime(compute_stop_time_ - compute_start_time_);
```

Comment:
I don't think this can work as `compute_stop_time_` isn't yet set at this time. It's only set after `compute()` has finished, but `storeSolution()` is called as part of `compute()`.
I think you need to get time here again.